### ISSUE TITLE
read delta table from GCP regression in 0.6.0

### ISSUE THREAD

**Issue by djouallah** [delta-io/delta-rs#778]
# Environment

**Delta-rs version**:0.6.0

**Binding**: Python 

**Environment**:
- **Cloud provider**: Colab
- **OS**:Linux
- **Other**:

***
# Bug

```
from deltalake import DeltaTable
dt = DeltaTable("gs://test_delta1/delta").to_pyarrow_dataset()
print ('number of files ' + str(len(dt.files)))
dt.head(1)
```
now generate this erros
PyDeltaTableError: Object at location gs:/test_delta1/delta/Date=2022-01-08/0-745c11c3-5354-4b89-b476-dba8dbdadf86-0.parquet not found: response error "No such object: test_delta1/gs:/test_delta1/delta/Date=2022-01-08/0-745c11c3-5354-4b89-b476-dba8dbdadf86-0.parquet", after 0 retries: HTTP status client error (404 Not Found) for url

**Comment by wjones127** [comment]
We just swapped out our GCS implementation, and need to setup automated integration testing for it. cc @roeap

**Comment by roeap** [comment]
I'll look into enabling integration tests for gcs.

**Comment by roeap** [comment]
Quick update. Testing our gcs backed object store was almost straight forward. Turns out there seems to be a bug upstream with the *_if_not_exists methods. Hopefully I can resolve that though since we need an upstream PR anyhow.

However this does indicate, that the error "might" be somewhere on the python side.

@djouallah could you see what happens if you do not convert the table to a pyarrow dataset and just print the version?

**Comment by djouallah** [comment]
@roeap 
![image](https://user-images.githubusercontent.com/12554469/187882667-a652fd68-027f-48c0-9406-098975305704.png)

**Comment by roeap** [comment]
We have some initial integration tests running (#779), So far this does not include the tests where we rely on if_not_exists behaviour. This should however already cover both the read and write path on the rust side.

So far my best guess is that somehow the paths on the python side got messed up when we did the switch to `object_store`.

**Comment by wjones127** [comment]
Once you have integration tests for Rust setup, I can work on having some Python-specific ones and should be able to fix this.

**Comment by roeap** [comment]
WE could just merge the PR currently in draft. In parallel I could see upstream to get the if_not_exists stuff fixed. But some integration tests are running and passing :)

**Comment by roeap** [comment]
@djouallah - we just released `0.6.1`, could you check if this gets resolved?

**Comment by djouallah** [comment]
thank you, I confirm it is working, very good job !!!
